paws_collector.js

```diff
@@ -498,17 +499,20 @@ class PawsCollector extends AlAwsCollector {
            
             if (err) {
                 /**
-                 * MS api return the logs but if build payload size > 10MB, and split logs array the in two part.
+                 * MS api return the logs but if build payload size > 10MB or if ingest return status code 307;
+                 * Split data per 10K message batches.
                  * Call the processLog again to send Logmsgs to ingest. 
                  */
-                if ((typeof err === 'string' && err.indexOf('Maximum payload size exceeded') !== -1)) {
-
-                    const half = Math.floor(logs.length / 2);
-                    const indexArray = [
-                        { start: 0, stop: half },
-                        { start: half, stop: logs.length }
-                    ];
-
+                if ((typeof err === 'string' && err.indexOf('Maximum payload size exceeded') !== -1) || (err.statusCode && err.statusCode === 307)) {
```

Comment:
I suggest we do split regardless of this `if` condition. Basically move the code below to [here](https://github.com/alertlogic/paws-collector/pull/205/files#diff-82fd638370f7c86cddbb67ebf81bc2952ba84a1e7dc12ce1ed0fe1da30045492R497)